cachito/web/content_manifest.py

```diff
@@ -85,9 +104,50 @@ def process_go_package(self, package, dependency):
         :param Dependency dependency: the go-package package dependency to process
         """
         if dependency.type == "go-package":
-            icm_dependency = {"purl": to_purl(dependency)}
+            if dependency.version and dependency.version.startswith("."):
+                dep_purl = self._get_local_go_package_dep_purl(package, dependency)
+            else:
+                dep_purl = to_purl(dependency)
+
+            icm_dependency = {"purl": dep_purl}
             self._gopkg_data[package]["dependencies"].append(icm_dependency)
 
+    def _get_local_go_package_dep_purl(self, package: "Package", dependency: "Package") -> str:
+        """
+        Get the purl for a local go-package dependency.
+
+        :param Package package: the go-package package to process
+        :param Dependency dependency: the go-package package dependency to process
+        """
+        if not dependency.version or not dependency.version.startswith("."):
```

Comment:
Why do we check this twice? https://github.com/containerbuildsystem/cachito/pull/773/commits/b292ecdff5200276c0202524ec38f62bb1d211ae#diff-191e36300d9d76bf9bd1c4a7329c01e1c11d9c8b12b018821d7399663a23cddeR107

Reply:
mypy wasn't pleased that version is optional in the Package class :smile:

Reply:
OK :)